native/java/org/jpype/JPypeContext.java

```diff
@@ -94,34 +94,40 @@ static public JPypeContext getInstance()
    * @param bootLoader is the classloader holding JPype resources.
    * @return the created context.
    */
-  public static JPypeContext createContext(long context, ClassLoader bootLoader)
+  static JPypeContext createContext(long context, ClassLoader bootLoader, String nativeLib)
   {
-    instance = new JPypeContext();
-
-    instance.context = context;
-    instance.bootLoader = bootLoader;
-    instance.typeFactory = new TypeFactoryNative();
-    instance.typeManager = new TypeManager(context, instance.typeFactory);
-    instance.typeManager.typeFactory = instance.typeFactory;
+    if (nativeLib != null)
+      System.load(nativeLib);
```

Comment:
what is the intention here?

Reply:
The current thunk system uses registerNatives to do all the name binding rather than the expected Java_ natives load mechanism.  It was required because of the rather strange way that JPype was booted up with loading a jar from the inside of shared library.  The usual way that a Java library is loading is to name all the entry points with a Java name mangling convention and then naming the so the correct way so that Java loads it automatically.  Unfortunately, we have to name our shared library to match Python naming so we have to manuallyloading your native library with a  System.load.  

When I started the porting Python from Java launch I suddenly needed 50 native entry points.  It is simply a huge burden to manually connect all the natives (which is not checked).   To make it manageable a switch was required.  We already had most of the methods named correctly, so the process was just rename the entry points, remove the register natives calls, and then make the final linkage.

The new load sequence is as follows.  I flipped the process around, rather than building a jar and compiling it into the shared library, instead we build a jar and call it a "Python" extension so it gets installed as the same place as _jpype.  We then have the module find its file location and pass it to the start routine so that the native sections are loaded in mass using System.load. 

The same mechanism is used to start Python from Java.  In that case we know where the jar is and we need to get our Python module.  

Sorry for not explaining it here.   Technically I am moving from our very unconventional loading method to a more standard one, but it does need some explanation as the process is still depending on two pieces that have naming conflicts.  Python requires the file to be named one way, Java requires another, so I have to do the last part of the link manually here.